FINERACT-1207: Clean up fineract-provider/config/swagger following in… (#1418)
apache/fineract: docs/developers/swagger/client.md

Generate Apache Fineract API Client
============

Apache Fineract supports client code generation using [OpenAPI Generator](https://openapi-generator.tech). It uses [OpenAPI Specification Version 3.0.3](https://swagger.io/specification/).

## Generate API Client

The API client is built as part of the overall Fineract Gradle build. If you want to save (maybe) some time you can try to execute just the following line in root directory of the project:

- Run `./gradlew :fineract-client:build`

> The client JAR can be found in `build/libs`.

Reviewer: ```suggestion
The client JAR can be found in `fineract-client/build/libs`.
```

Author: Done

> Note: just build fineract-client might or might not save you some time. There are still project module dependencies that might trigger a complete build.

Reviewer: ```suggestion
Note: Build only `fineract-client` may or may not actually save you some build time. There are still project module dependencies that might trigger a complete build.
```

Author: Done

## Customize Generated Code using Mustache Templates

> The OpenAPI code generator uses Mustache Templates for generating the client library. For additional customizations you can add/edit custom templates inside the `fineract-client/config/swagger/templates` folder. Note: we are not maintaining any custom templates in the Git repository. Files that are added to this folder are ignored by Git.

Reviewer: Oh, that's why the ignore. But... do we want this? Should we "support" that? Why? Should we just remove this entire section, to simplify things? I understand it's TECHNICALLY possible. I'm not sure we would want to deal with questions from anyone. @Grandolf49 any thoughts? @ptuomola any views?

Author: You tell me... I'm neither here nor their. Personally I tend more to explicitly providing generated clients like "normal" Java client, RxJava client, TypeScript client etc. (see SDK discussion). Otherwise people will start asking about their custom templates...

Reviewer: > You tell me... I'm neither here nor their. Personally I tend more to explicitly providing generated clients like "normal" Java client, RxJava client, TypeScript client etc. (see SDK discussion). Otherwise people will start asking about their custom templates...

Well, that makes 2 of us! Let's decide to remove the customizable templates. Our whole modularity project here is about people NOT forking and making local changes to Fineract (but be able to extend it), so why enable it here? I'd be happy to review and probably merge this PR once you have entirely removed the templates support.
